src/shared/utils/nightingale.ts

```diff
@@ -32,3 +33,80 @@ export const markBackground = (markedData: ProcessedFeature) => {
 export const markBorder =
   (nightingaleViewRange: NightingaleViewRange) => (datum: ProcessedFeature) =>
     withinRange(datum.start, datum.end, nightingaleViewRange);
+
+export const getZoomedInRange = (
+  features: ProcessedFeature[],
+  sequenceLength: number
+): NightingaleViewRange => {
+  if (sequenceLength <= AA_ZOOMED) {
+    return {
+      'display-start': 1,
+      'display-end': sequenceLength,
+    };
+  }
+  const OFFSET = 5;
+  const minStart = Math.min(...features.map((feature) => feature.start));
+  const displayStart =
+    // Right at the start
+    (minStart < OFFSET && minStart) ||
+    // Right at the end
+    (minStart > sequenceLength - OFFSET - AA_ZOOMED &&
+      sequenceLength - AA_ZOOMED) ||
+    // Everything else
+    minStart - OFFSET;
+
+  return {
+    'display-start': displayStart,
+    'display-end': displayStart + AA_ZOOMED,
+  };
+};
+
+export async function* linearTimed(
+  start: number,
+  end: number,
+  time: number,
+  steps: number
+): AsyncGenerator<number, void, unknown> {
+  const stepSize = (end - start) / steps;
+  // Note that sleep time is inaccurate below 50ms
+  const stepTime = time / steps;
+  for (let step = 1; step < steps; step += 1) {
+    // eslint-disable-next-line no-await-in-loop
+    await sleep(stepTime);
+    yield start + step * stepSize;
+  }
+  // Ensure the final value is exactly end
+  yield end;
+}
+
+export async function* rangeTimed(
```

Comment:
I feel an explanation in the comments would be more helpful when we refer to it in the future